src/driver-steelseries.c

```diff
@@ -65,6 +65,78 @@
 #define STEELSERIES_BUTTON_KBD			0x51
 #define STEELSERIES_BUTTON_CONSUMER		0x61
 
+struct steelseries_point {
+	struct list link;
+
+	struct ratbag_color color;	/* point color */
+	uint8_t pos;				/* relative position in the cycle */
+};
+
+struct steelseries_cycle {
+	uint8_t led_id; 			/* led id */
+	uint16_t duration;			/* cycle duration */
+	bool repeat;				/* if the cycle restarts automatically */
+	uint8_t trigger_buttons;	/* trigger button combination */
+	struct list points;			/* colors in the cycle */
+};
+
+static void
+create_cycle(struct steelseries_cycle *cycle)
```

Comment:
rename to create_led_cycle. Also please move it to just above steelseries_write_led_v2